Build HTML doc from within Colvars worktree

giacomofiorin opened the pull request.

jhenin commented:
The generated HTML pages contain the old links at the top:
Updated versions of this manual: [[GROMACS](https://colvars.github.io/colvars-refman-gromacs/colvars-refman-gromacs.html)] [[LAMMPS](https://colvars.github.io/colvars-refman-lammps/colvars-refman-lammps.html)] [[NAMD](https://colvars.github.io/colvars-refman-namd/colvars-refman-namd.html)] [[VMD](https://colvars.github.io/colvars-refman-vmd/colvars-refman-vmd.html)]

jhenin commented:
On my machine the Gromacs doc builds with sans serif fonts and contains these links:
Updated versions of this manual: [[GROMACS](https://colvars.github.io/colvars-refman-gromacs/colvars-refman-gromacs.html)] [[LAMMPS](https://colvars.github.io/colvars-refman-lammps/colvars-refman-lammps.html)] [[NAMD](https://colvars.github.io/colvars-refman-namd/colvars-refman-namd.html)] [[VMD](https://colvars.github.io/colvars-refman-vmd/colvars-refman-vmd.html)]

but the 3 others use serif and contain these links:
Updated versions of this manual: [[GROMACS](https://colvars.github.io/colvars-refman-gromacs/colvars-refman-gromacs.html)] [[LAMMPS](https://colvars.github.io/colvars-refman-lammps/colvars-refman-lammps.html)] [[NAMD](https://colvars.github.io/colvars-refman-namd/colvars-refman-namd.html)] [[VMD](https://colvars.github.io/colvars-refman-vmd/colvars-refman-vmd.html)]

The files on github.io don't suffer from those issues, I suppose due to software versions.

giacomofiorin commented:
> On my machine the Gromacs doc builds with sans serif fonts and contains these links: Updated versions of this manual: [[GROMACS](https://colvars.github.io/colvars-refman-gromacs/colvars-refman-gromacs.html)] [[LAMMPS](https://colvars.github.io/colvars-refman-lammps/colvars-refman-lammps.html)] [[NAMD](https://colvars.github.io/colvars-refman-namd/colvars-refman-namd.html)] [[VMD](https://colvars.github.io/colvars-refman-vmd/colvars-refman-vmd.html)]
> 
> but the 3 others use serif and contain these links: Updated versions of this manual: [[GROMACS](https://colvars.github.io/colvars-refman-gromacs/colvars-refman-gromacs.html)] [[LAMMPS](https://colvars.github.io/colvars-refman-lammps/colvars-refman-lammps.html)] [[NAMD](https://colvars.github.io/colvars-refman-namd/colvars-refman-namd.html)] [[VMD](https://colvars.github.io/colvars-refman-vmd/colvars-refman-vmd.html)]

The two sets of links are identical, and they look like they were built following what is in the `master` branch, i.e. using the Makefile from the `colvars.github.io`. I had initially removed the latter to avoid confusion, but that was premature because it broke the CI for existing branches so I reverted it a few hours ago.

I also just now added the `Makefile` itself as a dependency, to trigger a rebuild when it is changed on top of a non-empty worktree.

> 
> The files on github.io don't suffer from those issues, I suppose due to software versions.

If you think that that may be an issue, can you re-try using the same [container](https://github.com/Colvars/colvars/pkgs/container/devel-containers/137259711?tag=texlive)?

giacomofiorin commented:
I just pushed the updated doc pages to the website repo. It would be nice to do that automatically via Actions, but doing it manually is still quicker than looking that up.

jhenin commented:
Previously, we used the Makefile in the website repo. Did you write a script for that or is it fully manual? I have some more doc updates coming up.

giacomofiorin commented:
No I don't have a script but it would be very simple:
```
export COLVARS_WEBSITE_TREE=/path/to/website/repo
/path/to/container.sif make install
```
